pandas/core/reshape/concat.py

```diff
@@ -498,6 +498,8 @@ def get_result(self):
                 index, columns = self.new_axes
                 df = cons(data, index=index)
                 df.columns = columns
+                if self.names:
```

Comment:
This would change the behavior of the names keyword in concatenate in a user facing way. I myself have not used this (I think) and are not familiar with it, so can not judge if this is good?

What should happen if axis is 0? Should names be applied to the resulting Index?

We should at least document. Also probably should add this case for DataFrames if we want this, because this would differ now.

Please also add concatenate tests

Reply:
im finding in the groupby code that im having to pin columns.name following concat calls.  it would be an api change, but i think a good one

Reply:
No strong opinion either way. Just wanted to list the things we should think about when/if doing this